src/target/riscv/riscv-013.c

```diff
@@ -1503,15 +1503,20 @@ static void deinit_target(struct target *target)
 	info->version_specific = NULL;
 }
 
-static int set_haltgroup(struct target *target, bool *supported)
+typedef enum {
+	HALTGROUP,
+	RESUMEGROUP
+} grouptype_t;
+static int set_group(struct target *target, bool *supported, unsigned group, grouptype_t grouptype)
 {
-	uint32_t write = set_field(DM_DMCS2_HGWRITE, DM_DMCS2_GROUP, target->smp);
+	uint32_t write = set_field(DM_DMCS2_HGWRITE, DM_DMCS2_GROUP, group);
```

Comment:
It may be more readable to split this one into two lines, to make it clear to the reader that we're setting two fields, not just one. (And the compiler optimizes the assignments to `write` anyway.)

Also a good place to put an assertion for the `group` parameter (must not exceed 5 bits).

```suggestion
	assert(group <= 31);
	
	uint32_t write = 0;
	uint32_t write = set_field(write, DM_DMCS2_GROUP, group);	
	uint32_t write = set_field(write, DM_DMCS2_HGWRITE, 1);
```

Reply:
OK